packages/cadl-ranch-specs/http/azure/core/page/main.tsp

```diff
@@ -200,3 +201,411 @@ interface TwoModelsAsPageItem {
+        "collectionId": "string",
+        "transactionId": "string"
+      }
+    ]
+  }
+  """)
+@route("/ledger-entries")
+@put
+op ListPaginationLedgerEntries is global.Azure.Core.Foundations.ResourceList<
+  LedgerEntry,
+  ListPaginationLedgerEntryInput,
+  PagedLedgerEntries
+>;
+
+/**
+ * The interval
+ */
+union Interval {
+  /**
+   * A string representing a custom interval.
+   */
+  string,
+
+  PT5S: "PT5S",
+  PT10S: "PT10S",
+  PT1M: "PT1M",
+  PT5M: "PT5M",
+  PT1H: "PT1H",
+}
+
+model MetricDimensions {
+  /**
+   * The interval (i.e. timegrain) of the query.
+   */
+  @query
+  interval?: Interval;
+
+  /**
+   * Metric name
+   */
+  @query
+  metricName?: string;
+
+  /**
+   * Metric namespace to query metric definitions for.
+   */
+  @query
+  metricNamespace: string;
+
+  /**
+   *The timespan of the query.
+   */
+  @query
+  timespan?: string;
+}
+
+@resource("values")
+@parentResource(Dimension)
+model DimensionValueListItem {
+  /**
+   * The values of the dimension.
+   */
+  @key
+  value: string[];
+}
+
+@resource("dimensions")
+@parentResource(TestRuns)
+model Dimension {
+  @key
+  @path("name")
+  @visibility("query")
+  name: string;
+}
+
+/**
+ * Test run model
+ */
+@resource("test-runs")
+model TestRuns {
+  /**
+   * Unique test run name as identifier
+   */
+  @key
+  @path
+  @visibility("read")
+  testRunId: string;
+}
+
+@scenario
+@scenarioDoc("""
+  List the dimension values for the given metric dimension name.
+  """)
+interface AdditionalParameter {
+  /**
+   * List the dimension values for the given metric dimension name.
+   */
+  @route("/metric-dimensions")
+  ListMetricDimensionValues is global.Azure.Core.ResourceList<
+    DimensionValueListItem,
+    global.Azure.Core.Traits.QueryParametersTrait<MetricDimensions>
+  >;
+}
+
+model BatchRequestHeaders {
+  /**
+   * An OData $filter clause.
+   */
+  @query
+  $filter?: string;
+
+  /**
+   * An OData $select clause.
+   */
+  #suppress "deprecated" "Deprecated in next release"
+  @query({
+    format: "csv",
+  })
+  $select?: string[];
+
+  /**
+   * An OData $expand clause.
+   */
+  #suppress "deprecated" "Deprecated in next release"
+  @query({
+    format: "csv",
+  })
+  $expand?: string[];
+}
+
+/**
+ * A Pool in the Azure Batch service.
+ */
+model BatchPool {
+  /**
+   * The unique identifier for the Pool.
+   */
+  @visibility("read")
+  id?: string;
+
+  /**
+   * The display name for the Pool.
+   */
+  @visibility("read")
+  displayName?: string;
+
+  /**
+   * The URL of the Pool.
+   */
+  @visibility("read")
+  url?: string;
+}
+/**
+ * "The result of listing the Pools in an Account.
+ */
+@global.Azure.Core.pagedResult
+model BatchPoolListResult {
+  /**
+   * The list of Pools.
+   */
+  @global.Azure.Core.items
+  value?: BatchPool[];
+
+  /**
+   * The URL to get the next set of results.
+   */
+  #suppress "@azure-tools/typespec-azure-core/casing-style" "The names of Property types must use camelCase"
```

Comment:
why we are having this linter suppression here? I did not see a violation here. `nextLink` here follows the rule